### PR TITLE
Test string utilities

### LINKED ISSUE #587: [Testing] `packages/website/src/shared/lib/utilities/string.ts`
### User Story

As a developer, I want the codebase to be fully tested, so that I am sure that we are promoting high code quality by releasing stable code with no regressions.

### Description

This work, along with many other tickets, is to improve our testing code coverage by writing tests for the file in the title.

To view our current code coverage, run `pnpm test`. Then, open this file: `<project-root>/coverage/lcov-report/index.html`.

### Acceptance Criteria

1. The file in the title is fully tested, including statements, branches, functions, and lines.
2. All existing test suites pass.

### Technical References

- [Jest documentation](https://jestjs.io/docs/getting-started)
- [React Testing Library documentation](https://testing-library.com/docs/react-testing-library/intro/)
- Look into existing test suites

### Additional Information

N/A

### PR DESCRIPTION
**Pull request by glenflorendo**
### Description

This PR provides complete testing of the string utility functions.

```
  String Utilities
    isSubstring
      ✓ handles without case-sensitivity (5 ms)
      ✓ handles with case-sensitivity (1 ms)
```

### Related Issues

Resolves #587 

### Testing

- Ran existing testing suites
- Write new tests

```
-----------------------------------------------|---------|----------|---------|---------|-----------------------
File                                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s     
-----------------------------------------------|---------|----------|---------|---------|-----------------------
 src/shared/lib/utilities                      |   13.79 |     37.5 |   16.66 |      15 |                       
  string.ts                                    |     100 |      100 |     100 |     100 |                       
```

### Checklist

- [x] I have followed all conventions outlined in our documentation.
- [x] I have fully tested my changes and confirmed that all new and existing tests pass.
- [x] I have written meaningful commit messages for all changes.
- [x] I have linted and formatted my changes to follow the code style of this repository.
- [x] I have updated our documentation, accordingly.
- [x] I have checked currently opened pull requests to ensure that there are no pending pull request for the same changes or issues.
- [x] I have confirmed that this pull request fully meets the acceptance criteria for all related issues listed above.

### CODE REVIEW

File: packages/website/src/shared/lib/utilities/string.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { isSubstring } from "./string";
+
+describe("String Utilities", () => {
+  describe("isSubstring", () => {
+    it("handles without case-sensitivity", () => {
+      expect(isSubstring("", "")).toStrictEqual(true);
```
Comment (gibsonliketheguitar) [inline packages/website/src/shared/lib/utilities/string.test.ts:6]: nit. do we need toStrictEqual for booleans?

File: packages/website/src/shared/lib/utilities/string.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      expect(isSubstring("ABC", "a")).toStrictEqual(true);
+      expect(isSubstring("ABC", "b")).toStrictEqual(true);
+      expect(isSubstring("ABC", "c")).toStrictEqual(true);
+
+      expect(isSubstring("ABC", "A")).toStrictEqual(true);
+      expect(isSubstring("ABC", "B")).toStrictEqual(true);
+      expect(isSubstring("ABC", "C")).toStrictEqual(true);
+
+      expect(isSubstring("ABC", "d")).toStrictEqual(false);
+      expect(isSubstring("ABC", "e")).toStrictEqual(false);
```
Comment (gibsonliketheguitar) [inline packages/website/src/shared/lib/utilities/string.test.ts:17]: test are meticulously and exhaustive, a good thing. nit. However, cases like "e' does not add to test suite, it is duplicate of 'd'

**Review by gibsonliketheguitar — CHANGES_REQUESTED** [review]
Testcases are hard because of the amount. I haven't quite figure out a heuristic on thinking about them yet, and this is related to leetcode

We should think about what to test.

Currently we are testing one index at a time, while exhaustive, it does not add to the quality of test 

pseudo code for is validSubsequece
```
 Given A string, and B substring
 
 > Find the first char of B in A,
 > init j = 0
 > Iterate through B, and check A[i+j] match
           A    aBc
                 i  = 1
 Eg.      B =  bc
               j  = 0
 
   A[1] = B
   A[1 + 0] = b
```

a few missing test case for stings & substring

Input Edge Case
0. A or B missing
1. A string , B substring . if B len > A len return false

Possible State/Validation of condition

No case senstive
-  string A = aBc, and string B = aa false
-  string A = aBc, and string B = ab true
-  string A = aBc, and string B = abc true

For case Sensitives we need
- string A = aBc and string B = b true
- string A = aBc and string Bc = true
- string A = aBc and string BC = false

File: packages/website/src/shared/lib/utilities/string.test.ts (hunk 1 above)
Comment (glenflorendo) [reply packages/website/src/shared/lib/utilities/string.test.ts:6]: Probably not. Same thing with strings, I don't think it does any harm, but if you feel strongly otherwise, we can definitely change it.

File: packages/website/src/shared/lib/utilities/string.test.ts (hunk 2 above)
Comment (glenflorendo) [reply packages/website/src/shared/lib/utilities/string.test.ts:17]: Yeah, a lot of these tests are "duplicates" in the sense that the test the same thing. It doesn't add anything, but it's included for completeness and a way to give more than 1 example for a test case.

File: packages/website/src/shared/lib/utilities/string.test.ts (hunk 1 above)
Comment (gibsonliketheguitar) [reply packages/website/src/shared/lib/utilities/string.test.ts:6]: agree it's harmless, no changes needed
